### PR TITLE
Back/task metadata

### PR DESCRIPTION
**Pull request by MillarayVVS**
# Summary

Rename schema to metada in Tasks to get columns info with get Components.

## Type of change

- Refactoring.

## Changes

- Rename schema to metadata on Tasks.
- Adds get_metadata method
- get metadata with getComponents

## How to Test

Run ```pytest tests/back/api/test_components_api.py```

### CODE REVIEW

**Comment by cristian-tamblay** [comment]
Implementation looks good. I'm missing a test to check the method get_metadata() and we merge.

File: DashAI/back/tasks/base_task.py [hunk 1]
```diff
@@ -14,6 +14,27 @@ class BaseTask:
...
+        """Get metadata values for the current task
+
+        Returns:
+            Dict[str, Any]: Dictionary with the metadata
+        """
+        metadata = cls.metadata
+
+        # Extract class names
+        inputs_types = [i.__name__ for i in metadata["inputs_types"]]
+        outputs_types = [i.__name__ for i in metadata["outputs_types"]]
```
Comment (pbadillatorrealba) [inline DashAI/back/tasks/base_task.py]:
```suggestion
        inputs_types = [input_type.__name__ for input_type in metadata["inputs_types"]]
        outputs_types = [output_type.__name__ for output_type in metadata["outputs_types"]]
```

File: DashAI/back/tasks/base_task.py [hunk 2]
```diff
@@ -14,6 +14,27 @@ class BaseTask:
     def schema(self) -> Dict[str, Any]:
         raise NotImplementedError
 
+    @classmethod
+    def get_metadata(cls) -> Dict[str, Any]:
+        """Get metadata values for the current task
+
+        Returns:
+            Dict[str, Any]: Dictionary with the metadata
+        """
```
Comment (pbadillatorrealba) [inline DashAI/back/tasks/base_task.py]: Maybe add here a short description of what the metadata consists of (input/output types, cardinality)?

**Review by pbadillatorrealba — COMMENTED** [review]
Great change!

I left some mini comments, but it is ok :)

**Pull request merged**